modules/lazy-images/lazy-images.php

```diff
@@ -0,0 +1,158 @@
+		add_action( 'wp_enqueue_scripts', array( $this, 'enqueue_assets' ) );
+
+		// Do not lazy load avatar in admin bar
+		add_action( 'admin_bar_menu', array( $this, 'remove_filters' ), 0 );
+	}
+
+	public function setup_filters() {
+		add_filter( 'the_content', array( $this, 'add_image_placeholders' ), 99 ); // run this later, so other content filters have run, including image_add_wh on WP.com
+		add_filter( 'post_thumbnail_html', array( $this, 'add_image_placeholders' ), 11 );
+		add_filter( 'get_avatar', array( $this, 'add_image_placeholders' ), 11 );
+	}
+
+	public function remove_filters() {
+		remove_filter( 'the_content', array( $this, 'add_image_placeholders' ), 99 );
+		remove_filter( 'post_thumbnail_html', array( $this, 'add_image_placeholders' ), 11 );
+		remove_filter( 'get_avatar', array( $this, 'add_image_placeholders' ), 11 );
+	}
+
+	public function add_image_placeholders( $content ) {
+		// Don't lazyload for feeds, previews
+		if ( is_feed() || is_preview() ) {
+			return $content;
+		}
+
+		// Don't lazy-load if the content has already been run through previously
+		if ( false !== strpos( $content, 'data-lazy-src' ) ) {
+			return $content;
+		}
+
+		// Don't lazyload for amp-wp content
+		if ( function_exists( 'is_amp_endpoint' ) && is_amp_endpoint() ) {
+			return $content;
+		}
+
+		// This is a pretty simple regex, but it works
+		$content = preg_replace_callback( '#<(img)([^>]+?)(>(.*?)</\\1>|[\/]?>)#si', array( __CLASS__, 'process_image' ), $content );
+
+		return $content;
+	}
+
+	static function process_image( $matches ) {
+		$old_attributes_str = $matches[2];
+		$old_attributes_kses_hair = wp_kses_hair( $old_attributes_str, wp_allowed_protocols() );
+
+		if ( empty( $old_attributes_kses_hair['src'] ) ) {
+			return $matches[0];
+		}
+
+		$old_attributes = self::flatten_kses_hair_data( $old_attributes_kses_hair );
+		$new_attributes = $old_attributes;
+
+		// Set placeholder and lazy-src
+		$new_attributes['src'] = self::get_placeholder_image();
+		$new_attributes['data-lazy-src'] = $old_attributes['src'];
+
+		// Handle `srcset`
+		if ( ! empty( $new_attributes['srcset'] ) ) {
+			$new_attributes['data-lazy-srcset'] = $old_attributes['srcset'];
+			unset( $new_attributes['srcset'] );
+		}
+
+		// Handle `sizes`
+		if ( ! empty( $new_attributes['sizes'] ) ) {
+			$new_attributes['data-lazy-sizes'] = $old_attributes['sizes'];
+			unset( $new_attributes['sizes'] );
+		}
+
+		$new_attributes_str = self::build_attributes_string( $new_attributes );
+
+		return sprintf( '<img %1$s><noscript>%2$s</noscript>', $new_attributes_str, $matches[0] );
+	}
+
+	private static function get_placeholder_image() {
+		/**
+		 * Allows plugins and themes to modify the placeholder image.
+		 *
+		 * @module lazy-images
+		 *
+		 * @since 5.6.0
+		 *
+		 * @param string The URL to the placeholder image
+		 */
+		return apply_filters(
+			'lazyload_images_placeholder_image',
```

Comment:
I decided to use the same filters as the Lazy Load plugin to allow for easy updating for VIP and other users. Should we use something else though?

Reply:
I am torn. I guess we should keep them the same though. Sites switching from the existing plugin will experience brokenness/regressions otherwise, and people switching from other plugins won't care. Developers might think it's weird that we didn't prefix it with `jetpack_`, but only for about 5 seconds.

Reply:
It would be good to leave a comment why we choose not to prefix it with `jetpack_`